R/MSnExperiment-class.R

```diff
@@ -57,3 +82,64 @@ setMethod(
             "\n"
         )
 })
+
+#' @rdname MSnExperiment
+readMSnExperiment <- function(file, sampleData, backend = BackendMzR(),
+                              smoothed = NA, BPPARAM = bpparam()) {
+    ## if (missing(backend) || !inherits(backend))
+    if (missing(file) || length(file) == 0)
+        stop("Parameter 'file' is required")
+    if (!all(file.exists(file)))
+        stop("Input file(s) can not be found")
+    file <- normalizePath(file)
+    if (!missing(sampleData)) {
+        if (is.data.frame(sampleData))
+            sampleData <- DataFrame(sampleData)
+    } else {
+        sampleData <- DataFrame(sampleIdx = seq_along(file))
+    }
+    if (!is.logical(smoothed))
+        stop("smoothed should be a logical")
+    .read_file <- function(z, files, smoothed) {
```

Comment:
I don't like this kind of internal functions (because they are hard to test with unit test).

Reply:
If I'm moving the function out of the function I'll also have to find a better name...